tools/build_config/build.sh

```diff
@@ -178,15 +174,18 @@ function main() {
     print_package_versions
   fi
 
-  if ((analyze)); then
-    run_analysis "${HERBERT_ROOT}"
-  fi
-
-  if ((build)); then
+  if ((configure)); then
     warning_msg="Warning: Build directory ${build_dir} already exists.\n\
         This may not be a clean build."
     echo_and_run "mkdir ${build_dir}" || warning "${warning_msg}"
     run_configure "${build_dir}" "${build_config}" "${build_tests}" "${matlab_release}" "${cmake_flags}"
+  fi
+
+  if ((analyze)); then
+    run_analysis "${build_dir}"
+  fi
+
+  if ((build)); then
```

Comment:
To avoid us having to add the `--configure` flag when we want to build (which would make this script out-of-sync with the pwsh equivalent), would it be possible to add an `if ... fi` here and call `run_configure` if it's not been run yet? Checking for the existence of `${build_dir}/CMakeCache.txt` should be a sufficient.